### PR TITLE
plugins/gitblame: switch to mkNeovimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

**Comment by GaetanLepage** [comment]
I had to re-create a PR because I had originally used a branch on the main repo (by mistake) and thus couldn't force-push on it...

`helpers` -> `lib.nixvim` seems to work great !

File: plugins/git/gitblame.nix [hunk 1]
```diff
@@ -6,66 +6,133 @@
...
-  config =
-    let
-      setupOptions = {
-        enabled = cfg.enable;
-        message_template = cfg.messageTemplate;
-        date_format = cfg.dateFormat;
-        message_when_not_committed = cfg.messageWhenNotCommitted;
-        highlight_group = cfg.highlightGroup;
-        display_virtual_text = helpers.ifNonNull' cfg.displayVirtualText (
-          if cfg.displayVirtualText then 1 else 0
```
Comment (MattSturgeon) [inline plugins/git/gitblame.nix:54]: This used to be defined as a bool, but transformed into `enum [ 0 1 ]`. Looks like the new option is still a bool; is that correct?

File: plugins/git/gitblame.nix [hunk 2]
```diff
@@ -6,66 +6,133 @@
...
+    ];
+    delay = 1000;
+    virtual_text_column = 80;
+    use_blame_commit_file_urls = true;
   };
 
-  config =
-    let
-      setupOptions = {
-        enabled = cfg.enable;
```
Comment (MattSturgeon) [inline plugins/git/gitblame.nix:48]: Looks like we used to set `settings.enabled` by default. Is there a way we can warn that this is no longer the case?

File: plugins/git/gitblame.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/git/gitblame.nix:54]: Yes, the upstream option (in lua) is indeed `bool`: https://github.com/f-person/git-blame.nvim/blob/master/lua/gitblame/config.lua#L11

File: plugins/git/gitblame.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/git/gitblame.nix:48]: Anyway, the default behavior is still `true`. Now the user has the option to explicitly disable it.

File: plugins/git/gitblame.nix [hunk 3]
```diff
@@ -6,66 +6,133 @@
...
-      '';
+  extraOptions = {
+    gitPackage = nixvim.mkPackageOption {
+      name = "git";
+      default = pkgs.git;
     };
+  };
+
+  extraConfig = cfg: { extraPackages = [ cfg.gitPackage ]; };
+
```
Comment (MattSturgeon) [inline plugins/git/gitblame.nix:137]:
```suggestion
```

**Comment by traxys** [comment]
Could we try removing the `with lib;` too?

**Comment by MattSturgeon** [comment]
> Could we try removing the `with lib;` too?

I agree this may be nice going forward.

Although, in practice `with lib` is unlikely to cause us major issues?

If we only need a handful of things from `lib`, maybe do `inherit (lib) x y z`?

**Comment by GaetanLepage** [comment]
I made an attempt, please, let me hear your feedback.

File: plugins/git/gitblame.nix [hunk 4]
```diff
@@ -5,67 +5,135 @@
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.gitblame;
+  inherit (lib.nixvim) defaultNullOpts;
```
Comment (MattSturgeon) [inline plugins/git/gitblame.nix:9]: `inherit (lib) types` would also be ok IMO. Or even `types = lib.nixvim.nixvimTypes`?

Off topic: should we consider merging our types into `lib.types` (like we now do for maintainers)? Or is it better to keep our types under the `lib.nixvim` namespace?

**Review by MattSturgeon — APPROVED** [review]
No issues from me, but I'm out and about; reading from my phone screen.

Since it'd be easy for me to miss something maybe it's best if @traxys also review?

File: plugins/git/gitblame.nix (hunk 4 above)
Comment (GaetanLepage) [reply plugins/git/gitblame.nix:9]: > `inherit (lib) types` would also be ok IMO. Or even `types = lib.nixvim.nixvimTypes`?
> 
> Off topic: should we consider merging our types into `lib.types` (like we now do for maintainers)? Or is it better to keep our types under the `lib.nixvim` namespace?

Why not mixing them if the two sets don't clash.

Comment (MattSturgeon) [reply plugins/git/gitblame.nix]: The only reason I can think of is to be explicit about what is our custom stuff and what is from nixpkg's lib.

How important that is is up for debate. I think my preference would be to merge our types into `lib.types`.

**Comment by GaetanLepage** [comment]
Should we merge this then ?

**Comment by MattSturgeon** [comment]
> Should we merge this then ?

👍 from me

File: plugins/git/gitblame.nix (hunk 4 above)
Comment (MattSturgeon) [reply plugins/git/gitblame.nix]: See #1978

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**